src/bin/agent/CommandStart.ts

```diff
@@ -1,36 +1,229 @@
-      } else {
-        const agent = await PolykeyAgent.createPolykeyAgent({
-          password,
-          logger: this.logger.getChild(PolykeyAgent.name),
-          nodePath: options.nodePath,
-        });
+      // Handling exit signals.
+      let polykeyAgent: PolykeyAgent;
+      const handleSignal = async (_signal: NodeJS.Signals) => {
+        if (polykeyAgent != null) {
+          try {
+            await polykeyAgent.stop();
+            process.exitCode = 0;
+          } catch (e) {
+            if (e instanceof ErrorPolykey) {
+              process.stderr.write(
+                binUtils.outputFormatter({
+                  type: 'error',
+                  name: e.name,
+                  description: e.description,
+                  message: e.message,
+                }),
+              );
+              process.exitCode = e.exitCode;
+            } else {
+              // Unknown error, this should not happen
+              process.stderr.write(
+                binUtils.outputFormatter({
+                  type: 'error',
+                  name: e.name,
+                  description: e.message,
+                }),
+              );
+              process.exitCode = 255;
+            }
+          } finally {
+            process.exit(process.exitCode);
+          }
+        }
+      };
+      process.once('SIGINT', handleSignal);
+      process.once('SIGTERM', handleSignal);
+      process.once('SIGHUP', handleSignal);
 
-        // If started add handlers for terminating.
-        const termHandler = async () => {
-          await agent.stop();
+      // TODO: Fix up these functions
+      const agentConfig = {
+        nodePath: options.nodePath,
+        password: await binUtils.getPassword(options.passwordFile, this.fs),
+        keysConfig: {
+          recoveryCode: await binUtils.getRecoveryCode(
+            options.recoveryFile,
+            this.fs,
+          ),
+          fastKeyGen: process.env.PK_FAST_KEY_GEN === 'true',
+        },
+        networkConfig: {
+          clientHost: options.clientHost,
+          clientPort: options.clientPort,
+          ingressHost: options.ingressHost[0],
+          ingressPort: options.ingressPort,
+        },
+      };
+      try {
+        const recoveryCode = await bootstrapUtils.bootstrapState({
+          nodePath: agentConfig.nodePath,
+          password: agentConfig.password,
+          recoveryCode: agentConfig.keysConfig.recoveryCode,
+          fresh: options.fresh,
+          fs: this.fs,
+          logger: this.logger.getChild(bootstrapUtils.bootstrapState.name),
+        });
+        // TODO: this should only happen when we bootstrap
+        process.stdout.write(
+          binUtils.outputFormatter({
+            type: options.format === 'json' ? 'json' : 'list',
+            data: [
+              `Polykey was created from scratch.`,
+              `This is your recovery code, please save it in a safe place.`,
```

Comment:
These sorts of messages are supposed to go to `this.logger.info`. The STDOUT is meant to be used for data that can be later parsed by other programs. So I'm changing these human readable messages to be `this.logger.info` and then the output should be `process.stdout.write(recoveryCode + '\n');`.